website/scripts/build-static-content.js

```diff
@@ -200,36 +200,36 @@ module.exports = {
               mdString = mdString.replace(/(<call-to-action[\s\S]+[^>\n+])\n+(>)/g, '$1$2'); // « Removes any newlines that might exist before the closing `>` when the <call-to-action> compontent is added to markdown files.
+              mdString = mdString.replace(/(```)([a-zA-Z0-9\-]*)(\s*\n)/g, '$1\n' + '<!-- __LANG=%' + '$2' + '%__ -->' + '$3'); // « Based on the github-flavored markdown's language annotation, (e.g. ```js```) add a temporary marker to code blocks that can be parsed post-md-compilation when this is HTML.  Note: This is an HTML comment because it is easy to over-match and "accidentally" add it underneath each code block as well (being an HTML comment ensures it doesn't show up or break anything).  For more information, see https://github.com/uncletammy/doc-templater/blob/2969726b598b39aa78648c5379e4d9503b65685e/lib/compile-markdown-tree-from-remote-git-repo.js#L198-L202
               let htmlString = await sails.helpers.strings.toHtml(mdString);
               htmlString = (// « Add the appropriate class to the `<code>` based on the temporary "LANG" markers that were just added above
                 htmlString
                 .replace(// Interpret `js` as `javascript`
                   // $1     $2     $3   $4
-                  /(<code)([^>]*)(>\s*)(\s\&lt;!-- __LANG=\%js\%__ --\&gt;)\s*/gm,
+                  /(<code)([^>]*)(>\s*)(\&lt;!-- __LANG=\%js\%__ --\&gt;)\s*/gm,
                   '$1 class="javascript"$2$3'
                 )
                 .replace(// Interpret `sh` and `bash` as `bash`
                   // $1     $2     $3   $4
-                  /(<code)([^>]*)(>\s*)(\s\&lt;!-- __LANG=\%(bash|sh)\%__ --\&gt;)\s*/gm,
+                  /(<code)([^>]*)(>\s*)(\&lt;!-- __LANG=\%(bash|sh)\%__ --\&gt;)\s*/gm,
                   '$1 class="bash"$2$3'
                 )
                 .replace(// When unspecified, default to `text`
                   // $1     $2     $3   $4
-                  /(<code)([^>]*)(>\s*)(\s\&lt;!-- __LANG=\%\%__ --\&gt;)\s*/gm,
+                  /(<code)([^>]*)(>\s*)(\&lt;!-- __LANG=\%\%__ --\&gt;)\s*/gm,
                   '$1 class="nohighlight"$2$3'
                 )
-                .replace(// Nab the rest, leaving the code language as-is.
+                .replace(// Finally, nab the rest, leaving the code language as-is.
                   // $1     $2     $3   $4               $5    $6
-                  /(<code)([^>]*)(>\s*)(\s\&lt;!-- __LANG=\%)([^%]+)(\%__ --\&gt;)\s*/gm,
+                  /(<code)([^>]*)(>\s*)(\&lt;!-- __LANG=\%)([^%]+)(\%__ --\&gt;)\s*/gm,
```

Comment:
FYI: To future explorers: The above regex changes are actually reverting back to the former regexes before attempting to fix HTML rendering for code blocks inside of bullet points which have an extra newline above them.

Instead, we opted for the solution of writing markdown where code blocks nested inside bullet points are included directly after the bullet, indented properly, but with no extra newlines separating them.

See #6785 for context.